src/async.ts

```diff
@@ -0,0 +1,21 @@
+import { Robot, Response } from 'hubot';
+import { RobotWithClient } from './hackbot';
+
+export interface AsyncRobot extends RobotWithClient {

```

Comment:
Maybe future proof all of this and rename to `AugmentedRobot` for when even more features are added.